src/Install/Requirements.php

```diff
@@ -0,0 +1,129 @@
+    // Check extension version
+    $this->checkExtension();
+
+    // Check server version
+    $this->checkServer($phase);
+
+    return $this->requirements;
+  }
+
+/**
+ * Checks mongodb extension version.
+ */
+  public function checkExtension() {
+    if (extension_loaded('mongodb')) {
+      $this->extension_name = 'mongodb';
+      if (class_exists(MongoClient::class)) {
+        $this->requirements['mongodb_extension'] += array(
+          'value' => t('Properly replaced'),
+          'description' => $t('The mongo-php-adapter library and the ext-mongodb extension are present, replacing the ext-mongo extension.'),
+          'severity' => REQUIREMENT_OK,
+        );
+      }
+      else {
+        $this->requirements['mongodb_extension'] += array(
+          'value' => t('Insufficient'),
+          'description' => $t('The ext-mongodb extension is present, but the mongo-php-adapter is missing, so it cannot replace the old ext-mongo extension.'),
+          'severity' => REQUIREMENT_ERROR,
+        );
+      }
+    }
+    elseif (!extension_loaded('mongo')) {
+      if (class_exists(MongoClient::class)) {
+        $this->requirements['mongodb_extension'] += array(
+          'value' => t('Insufficient'),
+          'description' => $t('The mongo-php-adapter library is present, but the ext-mongodb is missing, so it cannot replace the old ext-mongo extension.'),
+          'severity' => REQUIREMENT_ERROR,
+        );
+      }
+      else {
+        $this->requirements['mongodb_extension'] += array(
+          'value' => t('Insufficient'),
+          'description' => $t('Neither the old ext-mongo nor the new ext-mongodb extension nor the mongo-php-adapter library are present.'),
+          'severity' => REQUIREMENT_ERROR,
+        );
+      }
+    }
+    else {
+      $this->extension_name = 'mongo';
+      if (class_exists(\Alcaeus\MongoDbAdapter\TypeInterface::class)) {
+        $this->requirements['mongodb_library_redundant'] = array(
+          'title' => 'mongo-php-adapter',
+          'value' => t('Redundant'),
+          'description' => t('The mongo-php-adapter library is installed while the old ext-mongo extension is present. This does not hurt, but is somewhat redundant.'),
+          'severity' => REQUIREMENT_WARNING,
+        );
+      }
+    }
+  }
+
+/**
+ * Checks mongodb server version.
+ */
+  public function checkServer($phase) {
+    $this->extension = new ReflectionExtension($this->extension_name);
+    $this->version = $this->extension->getVersion();
+    $this->requirements['mongodb_extension'] += array(
+      'value' => $this->version,
+      'severity' => REQUIREMENT_OK,
+    );
+
+    // Server versions prior to 1.3 do not support findAndModify command
+    // needed by mongodb_next_id() function.
+    $this->min_version = '1.3';
+
+    // During install, the module is not yet loaded by core.
+    if ($phase == 'install') {
+      require_once __DIR__ . '/../../../mongodb.module';
+    }
+    $this->db = mongodb();
```

Comment:
$this->db is only used locally: better make it a local variable (encapsulation).